pandas/tests/io/test_common.py

```diff
@@ -497,6 +497,23 @@ def test_warning_missing_utf_bom(self, encoding, compression_):
             with pytest.raises(UnicodeError, match=msg):
                 pd.read_csv(path, compression=compression_, encoding=encoding)
 
+    def test_chunk_splits_multibyte_char(self):
+        """
+        Chunk splits a multibyte character with memory_map=True
+
+        GH 43540
+        """
+        # DEFAULT_CHUNKSIZE = 262144, defined in parsers.pyx
+        df = pd.DataFrame(data=["a" * 127] * 2048)
+
+        # Put two-bytes utf-8 encoded character "ą" at the end of chunk
+        # utf-8 encoding of "ą" is b'\xc4\x85'
+        df.iloc[2047] = "a" * 127 + "ą"
+        df.to_csv("data/bug-gh43540.csv", index=False, header=False, encoding="utf-8")
+        dfr = pd.read_csv("data/bug-gh43540.csv", header=None, memory_map=True)
```

Comment:
Please use `with tm.ensure_clean() as path:` for the file handle.

Either add engine="c" (in case the default engine changes in the future) or (better) move the test to `pandas/tests/io/parser/test_encoding.py` and use the `all_parsers` fixture:

```
def test_chunk_splits_multibyte_char(all_parsers):
    parser = all_parsers
    parser.read_csv(...)
```

Reply:
Moved test to test_encoding and changed to `tm.ensure_clean()` like you suggested